contracts/WildcardSteward_v3.sol

```diff
@@ -353,13 +353,39 @@ contract WildcardSteward_v3 is Initializable {
     function changeReceivingBenefactor(
         uint256 tokenId,
         address payable _newReceivingBenefactor
-    ) public onlyReceivingBenefactorOrAdmin(tokenId) {
+    )
+        public
+        onlyReceivingBenefactorOrAdmin(tokenId)
+        updateBenefactorBalance(benefactors[tokenId])
+        updateBenefactorBalance(_newReceivingBenefactor)
+    {
         address oldBenfactor = benefactors[tokenId];
-        require(
-            oldBenfactor != _newReceivingBenefactor,
-            "Cannot change to same address"
-        );
+
+        require(oldBenfactor != _newReceivingBenefactor);
+        require(address(0) != _newReceivingBenefactor);
+
+        // Collect patronage from old and new benefactor before changing totalBenefactorTokenNumerator on both
+        uint256 scaledPrice = price[tokenId].mul(patronageNumerator[tokenId]);
+        totalBenefactorTokenNumerator[oldBenfactor] = totalBenefactorTokenNumerator[oldBenfactor]
+            .sub(scaledPrice);
+        totalBenefactorTokenNumerator[_newReceivingBenefactor] = totalBenefactorTokenNumerator[_newReceivingBenefactor]
+            .add(scaledPrice);
+
         benefactors[tokenId] = _newReceivingBenefactor;
+        // NB No fund exchanging here please!
+    }
+
+    // NB This function is if an organisation loses their keys etc..
+    // It will transfer their deposit to their new benefactor address
+    // It should only be called once all their tokens also changeReceivingBenefactor
+    function changeReceivingBenefactorDeposit(
+        address oldBenfactor,
+        address payable _newReceivingBenefactor
+    ) public onlyAdmin {
+        require(benefactorFunds[oldBenfactor] > 0);
```

Comment:
This ensures the benefactor exists, plus they also don;t have a credit balance, since deposit or credit now must be zero